google-cloud-logging/src/main/java/com/google/cloud/logging/MonitoredResourceHelper.java

```diff
@@ -0,0 +1,132 @@
+
+  private static void addResourceLabels(String resourceType,
+      MonitoredResource.Builder resourceBuilder) {
+    String[] labels;
+    switch (resourceType) {
+      case "gae_app_flex":
+        labels = new String[]{"module_id", "version_id", "zone", "instance_id"};
+        break;
+      case "gae_app_standard":
+        labels = new String[]{"module_id", "version_id"};
+        break;
+      case "container":
+        labels = new String[]{"cluster_name", "zone"};
+        break;
+      case "gce_instance":
+        labels = new String[]{"instance_id", "zone"};
+        break;
+      default:
+        labels = new String[]{};
+    }
+    for (String label : labels) {
+      addLabel(resourceBuilder, label);
+    }
+  }
+
+
+  /* Return a self-configured monitored resource.
+   */
+  static MonitoredResource getResource(String projectId, String resourceType) {
+    if (Strings.isNullOrEmpty(resourceType)) {
+      resourceType = getAutoDetectedResourceType();
+    }
+    // Currently, "gae_app" is the supported logging resource type, but we distinguish
+    // between "gae_app_flex", "gae_app_standard" to support zone id, instance id logging on flex VMs.
+    // This method trims "gae_app_flex", "gae_app_standard" to "gae_app"
+    String resourceName = (resourceType.startsWith("gae_app")) ? "gae_app" : resourceType;
+
+    MonitoredResource.Builder builder = com.google.cloud.MonitoredResource
+        .newBuilder(resourceName)
+        .addLabel("project_id", projectId);
+    addResourceLabels(resourceType, builder);
+    return builder.build();
+  }
+
+  /* Detect monitored resource type using environment variables, else return default
+   */
+  private static String getAutoDetectedResourceType() {
+    if (System.getenv("GAE_INSTANCE") != null) {
+      return "gae_app_flex";
+    }
+    if (System.getenv("KUBERNETES_SERVICE_HOST") != null) {
+      return "container";
+    }
+    if (System.getenv("GAE_MODULE") != null) {
+      return "gae_app_standard";
+    }
+    if (MetadataConfig.getInstanceId() != null) {
+      return "gce_instance";
+    }
+    return defaultResourceType;
+  }
+
+  protected static List<Enhancer> getResourceEnhancers() {
+    String resourceType = getAutoDetectedResourceType();
+    return EnhancerFactory.getEnhancers(resourceType);
+  }
+
+  private static void addLabel(MonitoredResource.Builder resourceBuilder, String name) {
+    String value;
+    switch (name) {
+      case "module_id":
+        value = getAppEngineModuleId();
+        break;
+      case "version_id":
+        value = getAppEngineVersionId();
+        break;
+      case "instance_name":
+        value = getAppEngineInstanceName();
+        break;
+      case "zone":
+        value = MetadataConfig.getZone();
+        break;
+      case "instance_id":
+        value = MetadataConfig.getInstanceId();
+        break;
+      case "cluster_name":
+        value = MetadataConfig.getClusterName();
```

Comment:
Do you miss a `break` here? IMO without it, value will be null when `name` is equal to "cluster_name"